web/api/web_api_v1.c

```diff
@@ -775,13 +775,44 @@ static inline void web_client_api_request_v1_info_summary_alarm_statuses(RRDHOST
 
 static inline void web_client_api_request_v1_info_mirrored_hosts(BUFFER *wb) {
     RRDHOST *rc;
+    BUFFER *guids = buffer_create(2048);
```

Comment:
This buffer is allocated and freed only so that there is a single loop over the hosts. But looping over the hosts is cheap - it is only walking down a linked list. Allocating and freeing the buffer is expensive as it involves syscalls. Why not two loops that both `buffer_sprintf(wb` instead of two buffers?

Reply:
OK I will change it. Yep it was done to go trough the list only once while lock is held. There is therefore small improvement during lock but as you correctly pointed out at cost outside of the critical section. This would be improved also by mempooling the buffers but that is another topic